cf-monitord/mon_network.c

```diff
@@ -202,42 +218,288 @@ void MonNetworkGatherData(double *cf_this)
 {
     ResetNetworkData();
 
-    char comm[PATH_MAX + 3] = {0};
-    strncpy(comm, VNETSTAT[VSYSTEMHARDCLASS], (sizeof(comm) - 1));
+    Item *in[ATTR] = {0};
+    Item *out[ATTR] = {0};
 
-    if (!FileCanOpen(comm, "r"))
+#ifdef __linux__
+    if ((access("/proc/net/tcp", R_OK) != 0) || !GetNetworkDataFromProcNet(cf_this, in, out))
+#endif
     {
-        Log(LOG_LEVEL_VERBOSE,
-            "Cannot open '%s', aborting gathering of network data (monitoring)",
-            comm);
-        return;
-    }
-    strcat(comm, " -an");
+        char comm[PATH_MAX + 3] = {0};
+        strncpy(comm, VNETSTAT[VSYSTEMHARDCLASS], (sizeof(comm) - 1));
 
-    FILE *pp;
-    if ((pp = cf_popen(comm, "r", true)) == NULL)
-    {
-        Log(LOG_LEVEL_VERBOSE,
-            "Opening '%s' failed, aborting gathering of network data (monitoring)",
-            comm);
-        return;
-    }
+        if (!FileCanOpen(comm, "r"))
+        {
+            Log(LOG_LEVEL_VERBOSE,
+                "Cannot open '%s', aborting gathering of network data (monitoring)",
+                comm);
+            return;
+        }
+        strcat(comm, " -an");
```

Comment:
So, it seems you made room for this in the buffer with the `+ 3`, but this string is 4 bytes, what am I missing?

Reply:
Also, `strncat` seems safer.

Reply:
> So, it seems you made room for this in the buffer with the `+ 3`, but this string is 4 bytes, what am I missing?

I was missing one byte. :)